test/torch/tensors/test_precision.py

```diff
@@ -30,14 +30,23 @@ def test_encode_decode(workers, parameter):
     assert (x == torch.tensor([0.1, 0.2, 0.3])).all()
 
 
-def test_fix_prec_registration(hook):
-    hook.local_worker.is_client_worker = False
+@pytest.mark.parametrize("use_context", [False, True])
+def test_fix_prec_registration(hook, use_context):
+    if not use_context:
```

Comment:
Q: Should I add new tests to validate that the context manager is working correctly? (and leave the old tests how they were?)
Another way is the one I added here - have a **use_context** parameter and depending on that use the python context or not (I do not like this method that much because it is hard to read).
Third method would be to duplicate the tests and in one test use the parameter explicitly and in another to use the context.

Reply:
Thanks!
So you can add a specific test to verify it works as expected, like this one
```
def <...>_test(...):
    assert hook.local_worker.is_client_worker == True
    with hook.local_worker.registration_enabled():
        hook.local_worker.is_client_worker == False
    assert hook.local_worker.is_client_worker == True

Reply:
And then change all the occurences of the tests where you have the old syntax `assert hook.local_worker.is_client_worker == True` to use context instead